drivers/entropy/entropy_mcux_css.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_cssv2
+
+#include "mcuxClCss_Rng.h"
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/entropy.h>
+#include <zephyr/init.h>
+#include <zephyr/random/rand32.h>
+
+static int entropy_mcux_css_get_entropy(const struct device *dev, uint8_t *buffer, uint16_t length)
+{
+	ARG_UNUSED(dev);
+	uint8_t status = 0;
+
+	MCUX_CSSL_FP_FUNCTION_CALL_BEGIN(result, token, mcuxClCss_Prng_GetRandom(buffer, length));
+	if ((MCUX_CSSL_FP_FUNCTION_CALLED(mcuxClCss_Prng_GetRandom) != token) ||
+	    (result != MCUXCLCSS_STATUS_OK)) {
+		status = -ENOMSG;
```

Comment:
Shouldn't it be `-EAGAIN` or maybe just `-EIO` ? I mean, we documented that we return -errno on error but `-EAGAIN` or `-EIO` looks more suitable here.

Reply:
sure